smartsim/entity/model.py

```diff
@@ -213,6 +219,134 @@ def params_to_args(self):
+                     tag="",
+                     inputs=None,
+                     outputs=None):
+        """A TF, TF-lite, PT, or ONNX model to load into the DB at runtime
+
+        Each ML Model added will be loaded into an
+        orchestrator (converged or not) prior to the execution
+        of this Model instance
+
+        One of either model (in memory representation) or model_path (file)
+        must be provided
+
+        :param name: key to store model under
+        :type name: str
+        :param model: model in memory
+        :type model: str, optional # TODO figure out what to type hint this as
+        :param model_path: serialized model
+        :type model_path: file path to model
+        :param backend: name of the backend (TORCH, TF, TFLITE, ONNX)
+        :type backend: str
+        :param device: name of device for execution, defaults to "CPU"
+        :type device: str, optional
+        :param batch_size: batch size for execution, defaults to 0
+        :type batch_size: int, optional
+        :param min_batch_size: minimum batch size for model execution, defaults to 0
+        :type min_batch_size: int, optional
+        :param tag: additional tag for model information, defaults to ""
+        :type tag: str, optional
+        :param inputs: model inputs (TF only), defaults to None
+        :type inputs: list[str], optional
+        :param outputs: model outupts (TF only), defaults to None
+        :type outputs: list[str], optional
+        """
+        db_model = DBModel(
+            name=name,
+            backend=backend,
+            model=model,
+            model_file=model_path,
+            device=device,
+            devices_per_node=devices_per_node,
+            batch_size=batch_size,
+            min_batch_size=min_batch_size,
+            tag=tag,
+            inputs=inputs,
+            outputs=outputs
+        )
+        self._db_models.append(db_model)
+
+    def add_script(self, name, script=None, script_path=None, device="CPU", devices_per_node=1):
+        """TorchScript to launch with this Model instance
+
+        Each script added to the model will be loaded into an
+        orchestrator (converged or not) prior to the execution
+        of this Model instance
+
+        Device selection is either "GPU" or "CPU". If many devices are
+        present, a number can be passed for specification e.g. "GPU:1".
+
+        Setting ``devices_per_node=N``, with N greater than one will result
+        in the model being stored in the first N devices of type ``device``.
+
+        One of either script (in memory string representation) or script_path (file)
+        must be provided
+
+        :param name: key to store script under
+        :type name: str
+        :param script: TorchScript code
+        :type script: str, optional
+        :param script_path: path to TorchScript code
+        :type script_path: str, optional
+        :param device: device for script execution, defaults to "CPU"
```

Comment:
Do we catch if a user puts in ``GPU:1`` and a ``devices_per_node`` of > 1? I see we already so some name checking but not sure if this case is caught.

Reply:
We do now